### ISSUE TITLE
Nested parent - grandparent selectors?

### ISSUE THREAD

**Issue by qmmr** [postcss/postcss-nested#14]
Currently, we're not able to reference the _grandparent_ of the current selector.

```css
.item {
    display: block;

    &__view {
        display: block;
    }

    &--edit {
        & &__view {
            display: none;
        }
    }
}
```

Output:
```css
.item {
    display: block
}
.item__view {
    display: block;
}
.item--edit .item--edit__view {
    display: none;
}
```

Can we have something like `&&` for grandparent selector? So that the following code

```css
.item {
    display: block;

    &__view {
        display: block;
    }

    &--edit {
        & &&__view {
            display: none;
        }
    }
}
```

Output:
```css
.item {
    display: block
}
.item__view {
    display: block;
}
.item--edit .item__view {
    display: none;
}
```

I've created a [branch for this in my fork](https://github.com/qmmr/postcss-nested/pull/1).

**Comment by ai** [comment]
It is look like overengeering for me. Many users will not know about `&&` and anyway will not understand your code.

Sometimes it is better to wrte selector by hand, because complicated cases with nestes and custom syntax is difficult to understand.

**Comment by ai** [comment]
I think it is better:

```css
.item {
    display: block;

    &__view {
        display: block;
    }

    &--edit {
    }

    &--edit &__view {
    }
```

**Comment by ai** [comment]
I will change me mind if:

1. You will show me case, when grandparent selector is really useful.
2. Many users will ask it (I will not close this issue to get feedback for some time).

If nobody will ask for it, you can anyway create a `postcss-nested-grandpa` fork plugin. I will mention you in Readme.md.

**Comment by toomuchdesign** [comment]
Hi @ai / @qmmr,
I was interested in this particular topic, so I wrote a specific POSTCSS plugin which introduce a special `^&` selector to reference any previous nested ancestor. It should produce an output ready to be processed with `postcss-nested`. So everything remains very optional and sticked to POSTCSS philosophy.

The name is [postcss-nested-ancestors](https://www.npmjs.com/package/postcss-nested-ancestors).

@qmmr use case might be solved by:
```
.item {
    display: block;

    &__view {
        display: block;
    }

    &--edit {
        ^&__view {
            display: none;
        }
    }
}
```
It's the first POSTCSS plugin I wrote, happy to hear any comment/suggestion. Cheers!
